storage/reads/aggregate_resultset_test.go

```diff
@@ -0,0 +1,168 @@
+	}
+}
+
+type mockCursorIterator struct{}
+
+func (i *mockCursorIterator) Next(ctx context.Context, req *cursors.CursorRequest) (cursors.Cursor, error) {
+	// TODO: this is the last little piece to mock out the entirety of the I/O
+	return &mockIntegerArrayCursor{}, nil
+}
+func (i *mockCursorIterator) Stats() cursors.CursorStats {
+	return cursors.CursorStats{ScannedBytes: 500, ScannedValues: 10}
+}
+
+type mockReadCursor struct {
+	rows  []reads.SeriesRow
+	index int64
+}
+
+func newMockReadCursor(keys ...string) mockReadCursor {
+	rows := make([]reads.SeriesRow, len(keys))
+	for i := range keys {
+		rows[i].Name, rows[i].SeriesTags = models.ParseKeyBytes([]byte(keys[i]))
+		rows[i].Tags = rows[i].SeriesTags.Clone()
+		rows[i].Tags.Set([]byte("_m"), rows[i].Name)
+		rows[i].Query = &mockCursorIterator{}
+	}
+
+	return mockReadCursor{rows: rows}
+}
+
+func (c *mockReadCursor) Next() *reads.SeriesRow {
+	if c.index == int64(len(c.rows)) {
+		return nil
+	}
+	row := c.rows[c.index]
+	c.index++
+	return &row
+}
+func (c *mockReadCursor) Close()     {}
+func (c *mockReadCursor) Err() error { return nil }
+
+// The stats from a WindowAggregateResultSet are retrieved from the cursor.
+func TestNewWindowAggregateResultSet_Stats(t *testing.T) {
+
+	newCursor := newMockReadCursor(
+		"clicks click=1 1",
+	)
+
+	request := datatypes.ReadWindowAggregateRequest{}
+	resultSet, err := reads.NewWindowAggregateResultSet(context.Background(), &request, &newCursor)
+
+	if err != nil {
+		t.Errorf("Error creating WindowAggregateResultSet: %s", err)
+	}
+
+	// If .Next() was never called, seriesRow is nil and stats are empty.
+	stats := resultSet.Stats()
+	if stats.ScannedBytes != 0 || stats.ScannedValues != 0 {
+		t.Errorf("Expected statistics to be empty")
+	}
+
+	resultSet.Next()
+	stats = resultSet.Stats()
+	if stats.ScannedBytes != 500 {
+		t.Errorf("Expected scanned bytes: %d got: %d", 500, stats.ScannedBytes)
+	}
+	if stats.ScannedValues != 10 {
+		t.Errorf("Expected scanned values: %d got: %d", 10, stats.ScannedValues)
+	}
+}
+
+// A count window aggregate is supported
+func TestNewWindowAggregateResultSet_Count(t *testing.T) {
+
+	newCursor := newMockReadCursor(
+		"clicks click=1 1",
+	)
+
+	request := datatypes.ReadWindowAggregateRequest{
+		Aggregate: []*datatypes.Aggregate{
+			&datatypes.Aggregate{Type: datatypes.AggregateTypeCount},
+		},
+		WindowEvery: 10,
+	}
+	resultSet, err := reads.NewWindowAggregateResultSet(context.Background(), &request, &newCursor)
+
+	if err != nil {
+		t.Errorf("Error creating WindowAggregateResultSet: %s", err)
+	}
+
+	if !resultSet.Next() {
+		t.Errorf("Unexpected: resultSet could not advance")
+	}
+	cursor := resultSet.Cursor()
+	if cursor == nil {
+		t.Errorf("Unexpected: cursor was nil")
+	}
+	integerArrayCursor := cursor.(cursors.IntegerArrayCursor)
+	integerArray := integerArrayCursor.Next()
+
+	if !reflect.DeepEqual(integerArray.Values, []int64{2, 5, 1}) {
```

Comment:
Can we also verify the timestamps?

Reply:
Yeah, I _just_ remembered I wanted to do that this morning (it was very late/early when I finally put this up).

Reply:
Yup, I think I want to have coverage for the window calculation.